micro-rdk/src/common/grpc.rs

```diff
@@ -1499,11 +1564,68 @@ where
                 Some(path) => path.as_str(),
                 None => return Err(GrpcError::RpcInvalidArgument),
             };
-            svc.process_request(path, msg);
+
+            let grpc = GrpcServerInner {
+                buffer: Some(BytesMut::with_capacity(GRPC_BUFFER_SIZE)),
+                robot: &svc.robot,
+                signaling_server: &svc.signaling_server,
+            };
+
+            let mut trailers = HeaderMap::new();
+            trailers.insert("grpc-status", "0".parse().unwrap());
+            struct UnfoldState {
+                trailers: HeaderMap,
+                #[allow(clippy::type_complexity)]
+                stream: Option<
+                    Pin<
+                        Box<
+                            dyn futures_lite::Stream<Item = Result<Bytes, ServerError>>
+                                + Sync
+                                + Send,
```

Comment:
The type system needs to know that the stream future we are putting in here is `Sync + Send` in order to let it be provided as the body. But my understanding is that the compiler can't derive that knowledge since `handle_request` returns a `dyn Stream`. So we need to be explicit about it.

Reply:
Do you mean the  `UnfoldState` struct, or the `Pin<Box<dyn Stream<Item=...>> + Sync + Send>` type? For the former, no, I'd prefer to keep the struct local to the function that uses it, since it doesn't serve a purpose anywhere else. For the future type, sure, that's fine by me.

Reply:
I did mean the former, but I see your point. I don't think defining the latter outside is useful (although I am surprised clippy didn't complain)